server/src/controllers/surveyController.test.ts

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Items } from '@azure/cosmos';
+import SurveyDBHandler from '../databases/handlers/surveyDBHandler';
+import { storeSurveyResult } from '../controllers/surveyController';
+import { getServerConfig } from '../utils/getConfig';
+
+describe('surveyResultController', () => {
+  let response;
+  let next;
+
+  beforeEach(() => {
+    process.env.VV_COSMOS_DB_ENDPOINT = 'https://testinghost/';
+    response = {
+      send: jest.fn().mockReturnThis(),
+      status: jest.fn().mockReturnThis()
+    } as any;
+    next = jest.fn();
+  });
+
+  afterEach(() => {
+    delete process.env.VV_COSMOS_DB_ENDPOINT;
+    jest.resetAllMocks();
+  });
+
+  test('Should success on upsert', async () => {
+    const inputData: any = {
+      sessionId: 'test_session_id',
+      callId: 'test_call_id',
+      acsUserId: 'test_acs_user_id',
+      response: true
+    };
+    const request: any = { body: inputData };
+    const mockedUpsert = jest.fn();
+    const mockedCosmosClient = {
+      database: jest.fn().mockImplementation(() => {
+        return {
+          container: jest.fn().mockImplementation(() => {
+            return {
+              items: {
+                upsert: mockedUpsert
+              }
+            };
+          })
+        };
+      })
+    };
+    const config = getServerConfig();
```

Comment:
To make this a pure unit test, I don't think we should call `getServerConfig` or any external dependency here. Unit tests should isolate the unit under test and avoid calling dependencies that can pollute the test environment. Especially true for `getServerConfig`, which depends on the machine's environment variables and/or the defaultConfig.json. `getServerConfig` can also change under the hood which will affect this test.

Instead, we can just create the config directly:
`const config = { ... }`

Reply:
Okay that make sense. I thought since we are mocking the `cosmosClient`, the config doesn't matter what it is. But I will update on it.